libraries/freertos_plus/standard/https/src/iot_https_client.c

```diff
@@ -710,45 +729,70 @@ static int _httpParserOnBodyCallback(http_parser * pHttpParser, const char * pLo
     _httpsResponse_t * pHttpsResponse = (_httpsResponse_t *)(pHttpParser->data);
     pHttpsResponse->parserState = PARSER_STATE_IN_BODY;
 
-    /* Has the user provided a buffer and is it large enough to fit the body? The
-    case of body buffer not being large enough can happen if the body was received
-    in the header buffer and the body buffer can not fit in all the body. */
-    if((pHttpsResponse->pBodyCur != NULL) && (pHttpsResponse->pBodyEnd - pHttpsResponse->pBodyCur > 0))
-    {
-        /* There are two scenarios when we need to copy data around:
-        1. Some or all of the response body may have been received in the header
-        buffer. If that is the case, we copy the response body received in the
-        header buffer to the user provided body buffer.
-        2. When we receive chunked header, the actual body is separated in
-        multiple chunks which are preceeded by length. For example, a chunked
-        body may look like:
-
-        7\r\n
-        Mozilla\r\n 
-        9\r\n
-        Developer\r\n
-        7\r\n
-        Network\r\n
-        0\r\n 
-        \r\n
-
-        In this case, we want the parsed body buffer to contain actual body only
-        (MozillaDeveloperNetwork in the above example).
+    if((pHttpsResponse->bufferProcessingState == PROCESSING_STATE_FILLING_HEADER_BUFFER) && (pHttpsResponse->isAsync))
```

Comment:
This needs a proper state diagram documentation for making sure this code is maintainable and other engineers in the team can work on this.

Reply:
Will work on this thanks.